File: 프로그래머스/2/42578. 의상/의상.py
from collections import Counter
from functools import reduce

def solution(clothes):
    count_by_type = Counter([type for _, type in clothes])
    
    # 각 종류별로 선택할 수 있는 옵션의 수(해당 종류의 의상 수 + 1)
    # 모든 종류에 대해 이 옵션의 수를 곱하기
    # 아무것도 입지 않는 경우 빼기 - 1
    answer = reduce(lambda acc, x: acc * (x + 1), count_by_type.values(), 1) - 1
    
    return answer

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
  const clothesCount = new Map();
  let combs = 1;

  clothes.forEach(([_, type]) => {
    clothesCount.set(type, (clothesCount.get(type) || 0) + 1);
  });

  for (const [_, cnt] of clothesCount) {
    combs *= cnt + 1;
  }

  return combs - 1;
}
